python/ray/ml/train/integrations/rllib/rl_trainer.py

```diff
@@ -0,0 +1,114 @@
+from ray.ml.config import ScalingConfig, RunConfig
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.trainer import Trainer, GenDataset
+from ray.rllib.agents.trainer import Trainer as RLLibTrainer
+from ray.rllib.utils.typing import PartialTrainerConfigDict, EnvType
+from ray.tune import Trainable, PlacementGroupFactory
+from ray.tune.logger import Logger
+from ray.tune.registry import get_trainable_cls
+from ray.tune.resources import Resources
+from ray.util.annotations import PublicAPI
+from ray.util.ml_utils.dict import merge_dicts
+
+
+@PublicAPI(stability="alpha")
+class RLTrainer(Trainer):
+    def __init__(
+        self,
+        algorithm: Union[str, Type[RLLibTrainer]],
+        scaling_config: Optional[ScalingConfig] = None,
+        run_config: Optional[RunConfig] = None,
+        datasets: Optional[Dict[str, GenDataset]] = None,
+        preprocessor: Optional[Preprocessor] = None,
+        resume_from_checkpoint: Optional[Checkpoint] = None,
+        **train_kwargs
+    ):
+        self._algorithm = algorithm
+        self._train_kwargs = train_kwargs
+
+        super().__init__(
+            scaling_config, run_config, datasets, preprocessor, resume_from_checkpoint
+        )
+
+    def _get_rllib_config(self, process_datasets: bool = False) -> Dict:
+        config = self._train_kwargs.get("param_space", {}).copy()
+        num_workers = self.scaling_config.get("num_workers")
+        if num_workers is not None:
+            config["num_workers"] = num_workers
+
+        worker_resources = self.scaling_config.get("resources_per_worker")
+        if worker_resources:
+            res = worker_resources.copy()
+            config["num_cpus_per_worker"] = res.pop("CPU", 1)
+            config["num_gpus_per_worker"] = res.pop("GPU", 0)
+            config["custom_resources_per_worker"] = res
+
+        use_gpu = self.scaling_config.get("use_gpu")
+        if use_gpu:
+            config["num_gpus"] = 1
+
+        trainer_resources = self.scaling_config.get("trainer_resources")
+        if trainer_resources:
+            config["num_cpus_for_driver"] = trainer_resources.get("CPU", 1)
+
+        if process_datasets:
+            self.preprocess_datasets()
+            # Up for discussion: If datasets is passed, should we always
+            # set the input config? Is the sampler config required here, too?
+            if self.datasets:
+                config["input"] = "dataset"
+                config["input_config"] = {
+                    "loader_fn": lambda: self.datasets["train"],
+                }
+
+        return config
+
+    def training_loop(self) -> None:
+        pass
+
+    def as_trainable(self) -> Type[Trainable]:
+        base_config = self._param_dict
+        trainer_cls = self.__class__
+
+        if isinstance(self._algorithm, str):
+            rllib_trainer = get_trainable_cls(self._algorithm)
+        else:
+            rllib_trainer = self._algorithm
+
+        class AIRRLTrainer(rllib_trainer):
```

Comment:
Isn't this name a bit confusing?
Is it actually more or less RLlibTrainable?

Reply:
It's an internal name and the `__name__` attribute is overwritten below, so this shouldn't matter too much